### PR TITLE
[FLASK] Update snap delineator

### LINKED ISSUE #18238: Implement updated snaps delineator
Component to wrap—and differentiate—third party UI using our growing library of predefined components. Currently used to display **transaction insights** and on our **custom dialogs**.

Design & specification: https://www.figma.com/file/diidnCHx9FXj5LDiA5DZZr/Snaps-features?node-id=627%3A42274&t=4axfyXOGXshsOMdO-1

### PR DESCRIPTION
**Pull request by GuillaumeRx**
## Explanation

This updates the Snap delineator per the figma designs linked in the issue.

It also contains a small fix to not limit the snap authorship component width (so it fits the under content width)

* Fixes #18238

## Screenshots/Screencaps
image

image

image

## Manual Testing Steps

- Try a Tx insight snap or a snap that uses `snap_dialog`

### CODE REVIEW

**Review by Mrtenz — CHANGES_REQUESTED** [review]

**Review by Mrtenz — DISMISSED** [review]

File: ui/components/app/flask/snap-authorship/snap-authorship.js [hunk 1]
```diff
@@ -59,7 +59,7 @@ const SnapAuthorship = ({ snapId, className }) => {
       paddingRight={4}
       borderRadius={BorderRadius.pill}
       display={DISPLAY.FLEX}
-      style={{ maxWidth: 'fit-content', width: '100%' }}
+      style={{ width: '100%' }}
```
Comment (NidhiKJha) [inline ui/components/app/flask/snap-authorship/snap-authorship.js:62]: We have added width prop to Box recently, I think we can use it here to remove the inline css
```suggestion
      width={BLOCK_SIZES.FULL}
```

File: ui/pages/confirmation/confirmation.js [hunk 2]
```diff
@@ -327,25 +333,46 @@ export default function ConfirmationPage({
...
-            />
-          </Box>
-        )}
+        {
+          ///: BEGIN:ONLY_INCLUDE_IN(flask)
+          !isSnapDialog &&
+            ///: END:ONLY_INCLUDE_IN
+            pendingConfirmation.origin === 'metamask' && (
+              <Box
+                alignItems="center"
```
Comment (NidhiKJha) [inline ui/pages/confirmation/confirmation.js:342]:
```suggestion
                alignItems={AlignItems.center}
```

**Review by david0xd — DISMISSED** [review]

File: ui/components/app/flask/snap-delineator/snap-delineator.js [hunk 3]
```diff
@@ -13,37 +12,63 @@ import {
...
+        isError ? BackgroundColor.errorMuted : BackgroundColor.backgroundDefault
+      }
     >
       <Box
         className="snap-delineator__header"
         alignItems={AlignItems.center}
-        backgroundColor={BackgroundColor.infoMuted}
-        paddingLeft={2}
-        paddingRight={2}
+        paddingLeft={1}
```
Comment (hmalik88) [inline ui/components/app/flask/snap-delineator/snap-delineator.js:46]: Can we squeeze these into just the `padding` property?

File: ui/pages/confirmation/confirmation.js [hunk 4]
```diff
@@ -327,25 +334,46 @@ export default function ConfirmationPage({
...
+          !isSnapDialog &&
+            ///: END:ONLY_INCLUDE_IN
+            pendingConfirmation.origin === 'metamask' && (
+              <Box
+                alignItems={AlignItems.center}
+                marginTop={1}
+                paddingTop={1}
+                paddingRight={4}
+                paddingLeft={4}
+                paddingBottom={4}
```
Comment (hmalik88) [inline ui/pages/confirmation/confirmation.js:348]: same as above, merge please

File: ui/pages/confirmation/confirmation.js [hunk 5]
```diff
@@ -327,25 +334,46 @@ export default function ConfirmationPage({
...
+        }
+        {
+          ///: BEGIN:ONLY_INCLUDE_IN(flask)
+          isSnapDialog && (
+            <Box
+              alignItems="center"
+              marginTop={4}
+              marginLeft={4}
+              marginRight={4}
+              marginBottom={4}
```
Comment (hmalik88) [inline ui/pages/confirmation/confirmation.js:369]: same as above, merge please

File: ui/pages/confirmation/confirmation.js (hunk 4 above)
Comment (GuillaumeRx) [reply ui/pages/confirmation/confirmation.js:348]: We cannot merge here, one of them is set to `1`

**Review by hmalik88 — APPROVED** [review]

**Review by david0xd — APPROVED** [review]

**Pull request merged**